cpu/esp32/Kconfig

```
@@ -93,4 +100,22 @@ config CPU_MODEL
 config CPU
     default "esp32" if CPU_FAM_ESP32
 
+menu "ESP32 configurations"
+    depends on TEST_KCONFIG
+    depends on HAS_ARCH_ESP32
+
+config MODULE_ESP_SPI_RAM
+    bool "SPI RAM support"
+    select MODULE_ESP_IDF_HEAP
```

```suggestion
    depends on HAS_ESP_SPI_RAM
    select MODULE_ESP_IDF_HEAP
```
This option should only be available for boards that provide SPI RAM. I know we didn't make this difference in `Makefile.dep`s by using `FEATURES_PROVIDED` and `FEATURES_REQUIRED`, but with the introduction of Kconfig we could solve this problem. The question is whether we should also enhance the handling of this feature  in `Makefile.dep`s with this change.

---

BTW, feature `HAS_ESP_SPI_RAM` could be modelled using `CPU_MODEL_ESP32_WROVER*` because SPI RAM is a feature of WROVER modules and all board that use the WROVER module also have SPI RAM. The only exception is the `esp32-ttgo-t-beam` board which has a ESP SoC and an external SPI RAM connected to it.

---

Sure, makes sense. I'll try to add it. As we are currently comparing to Makefile features should match, we'll have to add it on both places.

---

The feature is refactored in Makefile here: b2b370669d123187a8d61dbc721c38f6ca933a3c. The dependency added here 446c7824ee5dd401478ac458cb0e4bfba46322ec. And for Kconfig here 7194df0726a9296d4dbd6555b5b8323cd3a473f5